enterprise/backend/src/metabase_enterprise/sandbox/models/params/field_values.clj

```diff
@@ -0,0 +1,80 @@
+(ns metabase-enterprise.sandbox.models.params.field-values
+  (:require [clojure.core.memoize :as memoize]
+            [metabase-enterprise.enhancements.ee-strategy-impl :as ee-strategy-impl]
+            [metabase-enterprise.sandbox.api.table :as sandbox.api.table]
+            [metabase.api.common :as api]
+            [metabase.models.field :as field :refer [Field]]
+            [metabase.models.field-values :as field-values :refer [FieldValues]]
+            [metabase.models.params.field-values :as params.field-values]
+            [metabase.public-settings.metastore :as settings.metastore]
+            [metabase.util :as u]
+            [pretty.core :as pretty]
+            [toucan.db :as db]))
+
+(def ^:private ^{:arglist '([user-id last-updated field])} fetch-sandboxed-field-values*
+  (memoize/ttl
+   ;; use a custom key fn for memoization so two Field maps with different keys but the same `:id` still have the same
+   ;; cache key.
+   ^{::memoize/args-fn (fn [[current-user-id updated-at field]]
+                         [current-user-id updated-at (u/the-id field)])}
```

Comment:
should permissions go into this as well? If someone doesn't have full access, complains to their boss, they expand access, they wouldn't see the newly available field values for 30 days under this strategy.

Reply:
That's a good point. I'll tweak it